### PR TITLE
[Enhancement] Task support kill API

### LINKED ISSUE #8093: Task support kill API
## Enhancement
If a task runs for a long time, the user may need to kill the task manually, and this function is also required for materialized views. So it is necessary to provide a Kill API to support these requirements

### PR DESCRIPTION
**Pull request by Astralidea**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #8093

## Problem Summary(Required) ：

If a task runs for a long time, the user may need to kill the task manually, and this function is also required for materialized views. So it is necessary to provide a Kill API to support these requirements.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 28 / 39 (71.79%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/scheduler/TaskRunManager.java|3|8|37.50%|
|:large_blue_circle:|com/starrocks/scheduler/TaskManager.java|8|13|61.54%|
|:large_blue_circle:|com/starrocks/scheduler/TaskRun.java|16|17|94.12%|
|:large_blue_circle:|com/starrocks/scheduler/TaskRunBuilder.java|1|1|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/scheduler/TaskRun.java [hunk 1]
```diff
@@ -31,8 +31,13 @@ public class TaskRun implements Comparable<TaskRun> {
...
 
-    private ConnectContext ctx;
-
+    // The ConnectCtx brought over when the TaskRun is constructed is usually the task that the user manually
+    // penalizes and executes. For the periodical task, it is triggered by the system,
+    // so typically this value may be NULL.
+    private ConnectContext buildCtx;
+    // The ConnectCtx that actually executes the SQL statement is usually copied or constructed
+    // when TaskRun is executed.
+    private ConnectContext runCtx;
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/scheduler/TaskRun.java:40]: I think taskRun should be a background task, why it needs ConnectContext field?

Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/scheduler/TaskRun.java:40]: I think taskRun should be a background task, why it needs ConnectContext field?

Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/scheduler/TaskRun.java:40]: I think taskRun should be a background task, why it needs ConnectContext field?

Comment (ABingHuang) [reply fe/fe-core/src/main/java/com/starrocks/scheduler/TaskRun.java:40]: it is a bit confusing with two context.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 40 / 46 (86.96%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/scheduler/TaskManager.java|8|13|61.54%|
|:large_blue_circle:|com/starrocks/scheduler/TaskRunManager.java|7|8|87.50%|
|:large_blue_circle:|com/starrocks/scheduler/TaskRunBuilder.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/scheduler/TaskRun.java|24|24|100.00%|

**Comment by Astralidea** [comment]
run starrocks_be_build

**Comment by Astralidea** [comment]
run starrocks_be_unittest

**Comment by Astralidea** [comment]
run starrocks_clang-format

**Comment by Astralidea** [comment]
run starrocks_fe_unittest

**Comment by Astralidea** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 37 / 50 (74.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/scheduler/TaskRunManager.java|3|8|37.50%|
|:large_blue_circle:|com/starrocks/scheduler/TaskManager.java|8|13|61.54%|
|:large_blue_circle:|com/starrocks/scheduler/Task.java|6|8|75.00%|
|:large_blue_circle:|com/starrocks/scheduler/TaskRun.java|15|16|93.75%|
|:large_blue_circle:|com/starrocks/scheduler/persist/TaskRunStatus.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/scheduler/TaskBuilder.java|2|2|100.00%|

**Review by ABingHuang — APPROVED** [review]
LGTM

File: fe/fe-core/src/main/java/com/starrocks/scheduler/Task.java [hunk 2]
```diff
@@ -50,6 +51,15 @@ public class Task implements Writable {
     @SerializedName("source")
     private Constants.TaskSource source = Constants.TaskSource.CTAS;
 
+    // set default to ROOT is for compatibility
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/scheduler/Task.java:54]: The task framework is new, why need to consider compatibility?

Comment (Astralidea) [reply fe/fe-core/src/main/java/com/starrocks/scheduler/Task.java:54]: Task framework already build in 2.3 version it supported submit statement for async CTAS.

**Review by kangkaisen — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Pull request merged**